rln/src/wasm.rs

```diff
@@ -0,0 +1,108 @@
+use crate::public::RLN;
+use wasm_bindgen::prelude::*;
+extern crate web_sys;
+
+use js_sys::{BigInt as JsBigInt, Object, Uint8Array};
+use num_bigint::BigInt;
+
+#[wasm_bindgen]
+pub fn init_panic_hook() {
+    console_error_panic_hook::set_once();
+}
+
+#[allow(clippy::not_unsafe_ptr_arg_deref)]
+#[no_mangle]
+#[wasm_bindgen(js_name = newRLN)]
+pub fn wasm_new(tree_height: usize, zkey: Uint8Array, vk: Uint8Array) -> *mut RLN {
+    let rln = RLN::new_with_params(tree_height, zkey.to_vec(), vk.to_vec());
+    return Box::into_raw(Box::new(rln));
+}
+
+#[allow(clippy::not_unsafe_ptr_arg_deref)]
+#[wasm_bindgen(js_name = getSerializedRLNWitness)]
+pub fn wasm_get_serialized_rln_witness(ctx: *mut RLN, input: Uint8Array) -> Uint8Array {
+    let rln = unsafe { &mut *ctx };
+    let rln_witness = rln.get_serialized_rln_witness(&input.to_vec()[..]);
+
+    Uint8Array::from(&rln_witness[..])
+}
+
+#[allow(clippy::not_unsafe_ptr_arg_deref)]
+#[no_mangle]
+#[wasm_bindgen(js_name = insertMember)]
+pub fn wasm_set_next_leaf(ctx: *mut RLN, input: Uint8Array) -> Result<(), String> {
+    let rln = unsafe { &mut *ctx };
+    if rln.set_next_leaf(&input.to_vec()[..]).is_ok() {
+        Ok(())
+    } else {
+        Err("could not insert member into merkle tree".into())
+    }
+}
+
+#[allow(clippy::not_unsafe_ptr_arg_deref)]
+#[no_mangle]
+#[wasm_bindgen(js_name = RLNWitnessToJson)]
+pub fn rln_witness_to_json(ctx: *mut RLN, serialized_witness: Uint8Array) -> Object {
+    let rln = unsafe { &mut *ctx };
+    let inputs = rln
+        .get_rln_witness_json(&serialized_witness.to_vec()[..])
+        .unwrap();
+
+    let js_value = serde_wasm_bindgen::to_value(&inputs).unwrap();
+    let obj = Object::from_entries(&js_value);
+    obj.unwrap()
+}
+
+#[allow(clippy::not_unsafe_ptr_arg_deref)]
+#[no_mangle]
+#[wasm_bindgen]
+pub fn generate_rln_proof_with_witness(
```

Comment:
The difference between this function and what zerokit has to generate the proof is that it receives the witness calculator that was calculated on JS outside of zerokit/ark-circom